contracts/XYZSwapPair.sol

```diff
@@ -8,26 +8,42 @@ import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import "./libraries/MathExt.sol";
 import "./libraries/FeeFomula.sol";
 import "./libraries/ERC20Permit.sol";
-import "./interfaces/IXYZSwapPair.sol";
+
 import "./interfaces/IXYZSwapFactory.sol";
 import "./interfaces/IXYZSwapCallee.sol";
+import "./interfaces/IXYZSwapPair.sol";
 import "./VolumeTrendRecorder.sol";
 
 contract XYZSwapPair is IXYZSwapPair, ERC20Permit, ReentrancyGuard, VolumeTrendRecorder {
     using SafeMath for uint256;
     using SafeERC20 for IERC20;
 
+    uint256 internal constant MAX_UINT112 = 2**112 - 1;
+    uint256 internal constant BPS = 10000;
+
+    struct ReserveData {
+        uint256 reserve0;
+        uint256 reserve1;
+        uint256 vReserve0;
+        uint256 vReserve1; // only used when isAmpPool = true
+    }
+
     uint256 public constant MINIMUM_LIQUIDITY = 10**3;
 
     address public factory;
```

Comment:
Can change to `IYZSwapFactory`, then don't have to keep wrapping the factory =)

Reply:
ok